src/adapter/graphql/src/queries/datasets/dataset_endpoints.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use opendatafabric as odf;
+use url::Url;
+
+use crate::prelude::*;
+use crate::queries::*;
+
+pub struct DatasetEndpoints<'a> {
+    owner: &'a Account,
+    dataset_handle: odf::DatasetHandle,
+}
+
+#[Object]
+impl<'a> DatasetEndpoints<'a> {
+    #[graphql(skip)]
+    fn get_base_url() -> std::result::Result<Url, InternalError> {
+        // TODO: Use value from config not envvar
+        //       https://github.com/kamu-data/kamu-node/issues/45
+        let raw_base_url =
+            std::env::var("KAMU_BASE_URL").unwrap_or_else(|_| "http://127.0.0.1:8080".to_string());
+
+        Url::parse(&raw_base_url).int_err()
+    }
+
+    #[graphql(skip)]
+    pub fn new(owner: &'a Account, dataset_handle: odf::DatasetHandle) -> Self {
+        Self {
+            owner,
+            dataset_handle,
+        }
+    }
+
+    #[allow(clippy::unused_async)]
+    async fn web_link(&self) -> Result<LinkProtocolDesc> {
+        let base_url = Self::get_base_url()?;
+        let url = format!("{base_url}{}", self.dataset_handle.alias);
+
+        Ok(LinkProtocolDesc { url })
+    }
+
+    #[allow(clippy::unused_async)]
+    async fn cli(&self) -> Result<CliProtocolDesc> {
+        let base_url = Self::get_base_url()?;
+        let url = format!("{base_url}{}", self.dataset_handle.alias);
+
+        let push_command = format!("kamu push {url}");
+        let pull_command = format!(
+            "kamu pull {url} --as {}",
```

Comment:
I think we can omit the `--as {name}` part. When `--as` is not provided the `PullService` will try to guess the name from URL by taking the last path component which should already be a dataset name.

The shorted the URLs - the less scary they are for users.

Reply:
Done